### PR TITLE
Add doc for 2D & 3D navigationmesh generation

### PR DESCRIPTION
**Pull request by smix8**
Adds documentation how to create navigationmesh / navpolygons for 2D and 3D both with editor tools as well with scripts + some technical background stuff.

The warning at the beginning is still valid, there is so much to add to this page but I run out of time.

I guess this together with the othe open pr's should fix issue https://github.com/godotengine/godot/issues/32425 and https://github.com/godotengine/godot/issues/38204.

### CODE REVIEW

File: tutorials/navigation/navigation_using_navigationmeshes.rst [hunk 1]
```diff
@@ -0,0 +1,325 @@
+.. _doc_navigation_using_navigationmeshes:
+
+Using NavigationMeshes
+======================
+
+.. warning::
+
+    This chapter is a stub for linking and unfinished.
```
Comment (mhilbrunner) [inline tutorials/navigation/navigation_using_navigationmeshes.rst:8]:
```suggestion
    This chapter is a stub for linking and unfinished.
```

Thats a bit of a weird note IMO, what exactly is missing? Lots of docs pages could use fleshing out, and a lot of stuff is missing or outdated. IME, we don't leave similar notes everywhere. Open to hear opinions about this however, just not sure how useful it is.

File: tutorials/navigation/navigation_using_navigationmeshes.rst [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+
+
+For 2D NavigationPolygon resources are used to draw outline points in the editor. From these outline points the NavigationServer2D creates a mesh to upload navigation data to the NavigationServer.
+
+For 3D NavigationMesh resources are used. Instead of providing draw tools the 3D variant 
+provides an extensive amount of parameters to bake a navigation mesh directly from 3D source geometry.
+
+.. note::
+
+    Technically there is no hard restriction between 2D and 3D how to use the given toolsets to create flat navigation meshes. The 2D drawing tool can be used to create a flat 3D navmesh and the 3D baking tool can be used to parse flat 3D geometry into 2D appropriated navigationmeshes.
```
Comment (mhilbrunner) [inline tutorials/navigation/navigation_using_navigationmeshes.rst:199]:
```suggestion
    Technically there is no hard distinction between 2D and 3D on how to use the given toolsets to create flat navigation meshes. The 2D drawing tool can be used to create a flat 3D navmesh and the 3D baking tool can be used to parse flat 3D geometry into 2D appropriated navigationmeshes.
```

File: tutorials/navigation/navigation_using_navigationmeshes.rst (hunk 1 above)
Comment (smix8) [reply tutorials/navigation/navigation_using_navigationmeshes.rst]: removed it

File: tutorials/navigation/navigation_using_navigationmeshes.rst [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+actor perfectly with the ground use other means like physics.
+
+.. warning::
+
+    Meshes need to be triangulated to work as navigation meshes. Other mesh face formats like quad or ngon are not supported.
+
+NavigationMesh geometry parsing
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+NavigationMesh resources have the following properties and parameters to control the geometry parsing behavior:
```
[the remark this answers is not in the document]
Comment (smix8) [reply tutorials/navigation/navigation_using_navigationmeshes.rst]: removed the section as it already exists in the class doc

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thanks!

**Pull request merged**